core/src/main/java/com/scalar/db/transaction/consensuscommit/Coordinator.java

```diff
@@ -52,24 +62,86 @@ public class Coordinator {
   public Coordinator(DistributedStorage storage) {
     this.storage = storage;
     coordinatorNamespace = NAMESPACE;
+    keyManipulator = new CoordinatorGroupCommitKeyManipulator();
   }
 
   @SuppressFBWarnings("EI_EXPOSE_REP2")
   public Coordinator(DistributedStorage storage, ConsensusCommitConfig config) {
     this.storage = storage;
     coordinatorNamespace = config.getCoordinatorNamespace().orElse(NAMESPACE);
+    keyManipulator = new CoordinatorGroupCommitKeyManipulator();
   }
 
   public Optional<Coordinator.State> getState(String id) throws CoordinatorException {
+    if (keyManipulator.isFullKey(id)) {
+      // The following read operation order is important. Reading a coordinator state is likely to
+      // occur for lazy recovery. The likelihood of lazy recovery attempts would increase if a
+      // transaction scanned by another transaction executing lazy recovery is delayed. In the group
+      // commit, a delayed transaction is specified with a full transaction ID, not with a parent
+      // transaction ID. Therefore, looking up with the full transaction ID should be tried first
+      // to minimize read operations as much as possible.
+
+      // Scan with the full ID for a delayed group that contains only a single transaction.
```

Comment:
I [moved this logic](https://github.com/scalar-labs/scalardb/pull/1728/commits/dc5395653b325c71a4e2742eb8a8f15a0a6e3418#diff-8049e66d11b81cdd5a1dc82c4977b5336741bcee24dd7ddd54c6a6940f09497bR86-R105) for the group commit to `getStateForGroupCommit()` to make things clearer. 

> regarding the `Scan with the full ID`, should it be something like `Regard the ID as a full ID and scan with the ID` since we don't know if the given ID is a full ID or not.

Ah, whether the ID is a full ID is already checked. But I changed the argument name to a more clear one in the commit.